### PR TITLE
Update 2-Domain-Driven-Design.rst

### PR DESCRIPTION
**Pull request by kevinklaus**
Changed some unnecessarily gendered and biased language to gender-neutral and inclusive language.

### CODE REVIEW

File: Documentation/2-BasicPrinciples/2-Domain-Driven-Design.rst [hunk 1]
```diff
@@ -14,10 +14,10 @@ and the specifications were written but still the client was not satisfied at
 the end or in worst case the software hasn't solved his problems at all.
 
 
-If you develop software for a customer, you must first understand his problems,
+If you develop software for a customer, you must first understand their problems,
```
Comment (mbrodala) [inline Documentation/2-BasicPrinciples/2-Domain-Driven-Design.rst]: Should be `... for customers ...` then.

File: Documentation/2-BasicPrinciples/2-Domain-Driven-Design.rst [hunk 2]
```diff
@@ -14,10 +14,10 @@ and the specifications were written but still the client was not satisfied at
...
 so you can offer your customer a tailored solution. This problem area is called
 the application domain within the terms of domain-driven design. By knowing the
-application domain of the customer, understanding his problems and having them clearly in
+application domain of the customer, understanding their problems and having them clearly in
```
Comment (mbrodala) [inline Documentation/2-BasicPrinciples/2-Domain-Driven-Design.rst]: Same here.

**Comment by kevinklaus** [comment]
The use of they/their/theirs is fine to use with singular gender-neutral language, but your suggestions avoids the topic overall and is therefor quite smooth!

**Comment by mbrodala** [comment]
Thanks, nice and down to earth solution to the issue.

**Pull request merged**